test/components/tree-view.spec.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import { expect, ClientRenderer, sinon, simulate, waitFor } from 'test-drive-react';
+import { TreeView, TreeItem } from '../../src';
+import { TreeViewDemo, treeData } from '../../demo/tree-view-demo';
+import { TreeItemData } from '../../src/components/tree-view/tree-view';
+
+const treeView = 'TREE_VIEW';
+const treeItem = 'TREE_ITEM';
+
+const noop = (item: Object) => false;
+
+function getLabelsList(data: {label: string, children?: Object[]}): string[] {
+    return [data.label].concat(...(data.children || []).map(getLabelsList));
+}
+
+describe('<TreeView />', () => {
+    const clientRenderer = new ClientRenderer();
+    afterEach(() => clientRenderer.cleanup());
+
+    function getTreeItem(id: string) {
+        return `${treeItem}_${id}`;
+    }
+
+    it('renders a tree view with a few children, clicks ones of then', async () => {
+        const { select, waitForDom } = clientRenderer.render(<TreeViewDemo />);
+
+        await waitForDom(() => expect(select(treeView + '_DEMO')).to.be.present());
+
+        const allNodesLabels = treeData.map(getLabelsList).reduce((prev, next) => [...prev, ...next]);
```

Comment:
Let's have one function returning `allNodesLabel` to make the test more readable.